Shield.Communication/Services/Bluetooth.cs

```diff
@@ -48,40 +49,62 @@ public Bluetooth()
 
             try
             {
-                PeerFinder.AllowBluetooth = true;
-                PeerFinder.AllowWiFiDirect = true;
-                PeerFinder.DisplayName = "Virtual Shields";
-                PeerFinder.Role = PeerRole.Peer;
-                if (!isPrePairedDevice)
-                {
-                    PeerFinder.Start();
-                }
+                var devices = RfcommDeviceService.GetDeviceSelector(RfcommServiceId.SerialPort);
+                var peers = await DeviceInformation.FindAllAsync(devices);
 
-                var peers = await PeerFinder.FindAllPeersAsync();
                 var connections = new Connections();
                 foreach (var peer in peers)
                 {
-                    connections.Add(new Connection(peer.DisplayName, peer));
+                    connections.Add(new Connection(peer.Name, peer));
                 }
 
                 return connections;
             }
-            catch (Exception)
+            catch (Exception e)
             {
                 return null;
             }
-
         }
 
         public override async Task<bool> Connect(Connection newConnection)
         {
+            HostName hostName = null;
+            string remoteServiceName = null;
+
             var peer = newConnection.Source as PeerInformation;
-            var result = await Connect(peer.HostName);
-            await base.Connect(newConnection);
+            if (peer != null)
+            {
+                hostName = peer.HostName;
+                remoteServiceName = "1";
```

Comment:
thanks - actually, the caller spits it into three objects - only this type uses the "1" where others are more enumerated and identified. this 'might' not even be necessary anymore as it comes in from an enumeration... I have to double check with multiple BT units